### PR TITLE
Dual extruder

### PR DESCRIPTION
**Pull request by snapmaker-lixiaolong**
support dual extruder

### CODE REVIEW

File: Marlin/src/module/dual_extruder_print_head.cpp [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    canbus_g.PushSendStandardData(msgid, buf, index);
+  }
+}
+
+void ReportMotorCsStatus() {
+  uint8_t buf[8];
+  uint8_t index = 1;
+  uint16_t msgid = registryInstance.FuncId2MsgId(FUNC_SWITCH_EXTRUDER);
+  if (msgid != INVALID_VALUE) {
+    canbus_g.PushSendStandardData(msgid, buf, index);
```
Comment (xhycode) [inline Marlin/src/module/dual_extruder_print_head.cpp:106]: 'buf' is not assigned a value

File: Marlin/src/HAL/hal_gpio.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  RCC_APB2Periph_GPIOE,
+  RCC_APB2Periph_GPIOF,
+  RCC_APB2Periph_GPIOG
+};
+
+void GpioRemap() {
+  GPIO_PinRemapConfig(GPIO_Remap_SWJ_JTAGDisable, ENABLE);
+}
+
+void GpioInit(uint8_t Port, GPIOMode_TypeDef Mode) {
```
Comment (scotthsl) [inline Marlin/src/HAL/hal_gpio.cpp:47]: Firstly, the  formal parameter 'Port', it actually indicates the 'Pin' number here，so should name it to 'pin'.
Secondly, per the code style, the  name of formal parameter should start with lower case

File: Marlin/src/HAL/hal_gpio.cpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  GPIO_InitTypeDef GPIO_InitStruct;
+  
+  RCC_APB2PeriphClockCmd(RCCTable[Port / 16], ENABLE);
+  GPIO_InitStruct.GPIO_Mode = Mode;
+  GPIO_InitStruct.GPIO_Pin = 1 << (Port % 16);
+  GPIO_InitStruct.GPIO_Speed = GPIO_Speed_50MHz;
+  GPIO_Init((GPIO_TypeDef*)PORTs[Port / 16], &GPIO_InitStruct);
+}
+
+void GpioWrite(uint8_t Port, uint8_t IOLevel) {
```
Comment (scotthsl) [inline Marlin/src/HAL/hal_gpio.cpp:57]: the  name of formal parameter should start with lower case

File: Marlin/src/HAL/hal_gpio.cpp [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  GPIO_TypeDef *pPort;
+  uint16_t Bits;
+  pPort = (GPIO_TypeDef*)PORTs[Port /16];
+  Bits = (1 << (Port % 16));
+  if(IOLevel) pPort->BSRR = Bits;
+  else pPort->BRR = Bits;
+}
+
+uint8_t GpioRead(uint8_t Port) {
```
Comment (scotthsl) [inline Marlin/src/HAL/hal_gpio.cpp:66]: the  name of formal parameter should start with lower case

**Pull request closed without merging**